src/main/java/net/fabricmc/loader/discovery/ModResolver.java

```diff
@@ -462,11 +462,13 @@ protected void compute() {
 			try (InputStream stream = Files.newInputStream(modJson)) {
 				info = ModMetadataParser.getMods(loader, stream);
 			} catch (JsonParseException e) {
-				throw new RuntimeException("Mod at '" + path + "' has an invalid fabric.mod.json file!", e);
+				throw new RuntimeException(String.format("Mod at \"%s\" has an invalid fabric.mod.json file!", path), e);
 			} catch (NoSuchFileException e) {
 				info = new LoaderModMetadata[0];
 			} catch (IOException e) {
-				throw new RuntimeException("Failed to open fabric.mod.json for mod at '" + path + "'!", e);
+				throw new RuntimeException(String.format("Failed to open fabric.mod.json for mod at \"%s\"!", path), e);
+			} catch (Throwable t) {
+				throw new RuntimeException(String.format("Failed to parse mod metadata for mod at \"%s\"", path), t);
```

Comment:
Imo Errors should not be rethrown as runtime exceptions.

Reply:
All of these exceptions kill the game though (or at least prevent it from being started normally) - would it make much of a difference to handle `Error`s differently to `Exception`s?

Reply:
(Plus any exception thrown here is ultimately wrapped in a `ModResolutionException` *anyway*).

Reply:
Well one of the exceptions just sets the mod metadata array to be empty (`NoSuchFileException`)

Reply:
Yes, but as that's a checked exception it can only be thrown by `Files.newInputStream()` so it can't be thrown by this method (which means I think we can ignore it for this).